core/src/main/scala/org/apache/spark/sql/streaming/eventhubs/EventHubsSource.scala

```diff
@@ -277,7 +277,8 @@ private[spark] class EventHubsSource(
               eventData.getProperties.asScala.getOrElse(k, "").toString
             })
           } else {
-            Seq(eventData.getProperties.asScala.map { case (k, v) => k -> v.toString })
+            Seq(eventData.getProperties.asScala.map { case (k, v) =>
+              k -> (if (v == null) "" else v.toString) })
```

Comment:
would you please add a unit test and also try if Spark can serialize null value (I think it can ), if Spark can serialize it, we do not even need to do any translation